gluten-core/src/main/scala/org/apache/gluten/extension/columnar/rewrite/PullOutPreProject.scala

```diff
@@ -158,9 +158,11 @@ object PullOutPreProject extends RewriteSingleNode with PullOutProjectHelper {
       val newAgg = copyBaseAggregateExec(agg)(
         newGroupingExpressions = newGroupingExpressions,
         newAggregateExpressions = newAggregateExpressions)
-      val preProject = ProjectExec(
-        eliminateProjectList(agg.child.outputSet, expressionMap.values.toSeq),
-        agg.child)
+      // ISSUE-5852: literals with same names are lost in expressionMap, need addback to Project
```

Comment:
Was the modification of the logic in this code unnecessary in your earliest version?

Reply:
Yes, the literals is not in the project outputs, it will case fallback.